deepchem/models/tensorgraph/layers.py

```diff
@@ -480,6 +482,24 @@ def _get_scope_name(self):
       return self._shared_with._get_scope_name()
 
 
+class BPGather2(Layer):
+
+  def __init__(self, **kwargs):
+    super(BPGather2, self).__init__(**kwargs)
+
+  def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
+    """ Merge features together """
+
+    last_dense, atom_feats = self._get_input_tensors(in_layers)
+    regression = tf.reduce_sum(last_dense, -1, keep_dims=True)
+    flags = tf.cast(tf.sign(atom_feats[:, :, 0]), tf.float32)
```

Comment:
I don't think this is actually the atom type.  I think this will end up being the 0th feature of the atom which will always be poisitve....

Reply:
Yeah you still need atom types for this masking.